From 801cc338356e0f0fd8728bf13e61a3c09c15b061 Mon Sep 17 00:00:00 2001
From: Tianshu Qiu <tian.shu.qiu@intel.com>
Date: Tue, 15 May 2018 14:13:18 +0800
Subject: [PATCH 1900/2328] media: intel-ipu4: imx355: Remove kernel version
 checking

Message for Open Source:
Remove kernel version checking

Message for Internal:
[Issue/Feature] Kernel version checking is not necessary as
the driver here is only used on kernel 4.14 and later.
[Root Cause/Changes] Remove kernel version checking

Change-Id: Ie4ff1e308153b8a5f151ac2338c745a64af1c3a1
Signed-off-by: Tianshu Qiu <tian.shu.qiu@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 drivers/media/i2c/imx355.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/drivers/media/i2c/imx355.c b/drivers/media/i2c/imx355.c
index 8f90416..5e4b149 100644
--- a/drivers/media/i2c/imx355.c
+++ b/drivers/media/i2c/imx355.c
@@ -1724,19 +1724,11 @@ static int imx355_probe(struct i2c_client *client,
 	imx355->sd.internal_ops = &imx355_internal_ops;
 	imx355->sd.flags |= V4L2_SUBDEV_FL_HAS_DEVNODE;
 	imx355->sd.entity.ops = &imx355_subdev_entity_ops;
-#if LINUX_VERSION_CODE > KERNEL_VERSION(4, 4, 0)
 	imx355->sd.entity.function = MEDIA_ENT_F_CAM_SENSOR;
-#else
-	imx355->sd.entity.type = MEDIA_ENT_T_V4L2_SUBDEV_SENSOR;
-#endif
 
 	/* Initialize source pad */
 	imx355->pad.flags = MEDIA_PAD_FL_SOURCE;
-#if LINUX_VERSION_CODE > KERNEL_VERSION(4, 4, 0)
 	ret = media_entity_pads_init(&imx355->sd.entity, 1, &imx355->pad);
-#else
-	ret = media_entity_init(&imx355->sd.entity, 1, &imx355->pad, 0);
-#endif
 	if (ret) {
 		dev_err(&client->dev, "%s failed:%d\n", __func__, ret);
 		goto error_handler_free;
-- 
2.7.4

